backend/src/seeds/seed.user.ts

```diff
@@ -46,20 +47,34 @@ class UserFactory extends Factory<User> {
       user.trainee = traineesFactory.generate();
     }
 
+    if (userType === 'instructor') {
+      user.instructor = instructorsFactory.generate();
+    }
+
     this.entities.push(user);
 
     return user;
   }
 
-  public generateFromData({ trainee, ...data }: Partial<User>) {
+  public generateFromData({ trainee, instructor, ...data }: Partial<User>) {
     const user = super.generateFromData(data);
 
-    if (trainee === null) {
-      traineesFactory.remove(user.trainee);
-    } else if (trainee !== undefined) {
+    if (user.trainee !== null) {
+      traineesFactory.remove(user.trainee!);
+      user.trainee = null;
+    }
+    if (trainee !== undefined && trainee !== null) {
       user.trainee = trainee;
     }
 
+    if (user.instructor !== null) {
+      instructorsFactory.remove(user.instructor!);
```

Comment:
As we're checking if the trainee is null there should be need to use Non-Null Assertion Operator (!)
```suggestion
      instructorsFactory.remove(user.instructor);
```